getnacelle/nacelle-react
# Update 'use-cart' hook to accept product metafields

Fixes #29 

- [react-hooks|use-cart]: Update 'use-cart' hook to accept product metafields
- This change updates the 'formatCartItem' to receive product metafields and pass them along
to the cart.
- There are some merchants that may need to pass this information if they are using integrations
such as Recharge, metafields should exist in the CartItem so the Hail Frequency API could process
the checkout with the right implementation.

### Type of Change


### What is being changed and why?
The object returned by the `formatCartItem` function has been updated to include the metafields, it also updates the variant object.

### How to Test
1. Add a product to the cart.
2. Inspect the cart item component to see the metafields are now there
![image](https://user-images.githubusercontent.com/2782816/105561154-9f970580-5cdb-11eb-8e3f-5b182446d89d.png)

2. Click checkout
3. You should see the Recharge checkout page and not the Shopify one, notice the URL with the Recharge domain
`https://checkout.rechargeapps.com/r/checkout/0d96f1d9a3?myshopify_domain=store.myshopify.com`
